api/lib/infrastructure/utils/ods/write-ods-utils.js

```diff
@@ -37,6 +37,25 @@ function updateXmlRows({ stringifiedXml, rowMarkerPlaceholder, rowTemplateValues
   return _buildStringifiedXmlFromXmlDom(parsedXmlDom);
 }
 
+function incrementRowsColumnSpan({ stringifiedXml, from, to, increment }) {
+  const parsedXmlDom = _buildXmlDomFromXmlString(stringifiedXml);
+  const rows = Array.from(parsedXmlDom.getElementsByTagName('table:table-row'));
+
+  for (let i = from; i < to; i++) {
+    const element = Array.from(rows[i].getElementsByTagName('table:table-cell'))
+      .reverse()
+      .find((element) => element.hasAttribute('table:number-columns-spanned'));
+    if (element) {
+      element.setAttribute(
+        'table:number-columns-spanned',
+        parseInt(element.getAttribute('table:number-columns-spanned')) + increment
+      );
+    }
+  }
```

Comment:
nitpick: (le for pourrait être utilisé pour éviter le reverse)
```suggestion
rows[i].forEach(row => {
    const element = Array.from(row.getElementsByTagName('table:table-cell'))
      .reverse()
      .find((element) => element.hasAttribute('table:number-columns-spanned'));
    if (element) {
      element.setAttribute(
        'table:number-columns-spanned',
        parseInt(element.getAttribute('table:number-columns-spanned')) + increment
      );
    }
 });
```

Reply:
Je comprends pas :).

Reply:
J'ai melangé 2 probleme. La je propose de remplacer la boucle par un `forEach`